src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -248,6 +225,47 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 						count = 0;
 					}
 				}
+
+				if (apkInputPathExists) {
```

Comment:
Can this entire block not be *refactored into a separate method*?  The levels of indentation here is getting…high.

Reply:
Done